metashare/api/jobs.py

```diff
@@ -0,0 +1,168 @@
+import contextlib
+import logging
+import traceback
+
+from asgiref.sync import async_to_sync
+from channels.layers import get_channel_layer
+from cumulusci.core.config import BaseProjectConfig, ScratchOrgConfig
+from cumulusci.core.runtime import BaseCumulusCI
+from django.utils.text import slugify
+from django.utils.timezone import now
+from django_rq import job
+from github3 import login
+from github3.exceptions import UnprocessableEntity
+
+from .github_context import (
+    extract_owner_and_repo,
+    get_cumulus_prefix,
+    local_github_checkout,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def report_scratch_org_error(instance, message):
+    from .serializers import ScratchOrgSerializer
+
+    message = {
+        "type": "SCRATCH_ORG_PROVISION_FAILED",
+        "payload": {"message": message, "model": ScratchOrgSerializer(instance).data},
+    }
+    model_name = instance._meta.model_name
+    id = str(instance.id)
+    group_name = "{model}.{id}".format(model=model_name, id=id)
+    channel_layer = get_channel_layer()
+    sent_message = {"type": "notify", "group": group_name, "content": message}
+    async_to_sync(channel_layer.group_send(group_name, sent_message))
+
+
+@contextlib.contextmanager
+def report_errors_on(scratch_org):
+    try:
+        yield
+    except Exception as e:
+        report_scratch_org_error(scratch_org, str(e))
+        tb = traceback.format_exc()
+        logger.error(tb)
+        raise
```

Comment:
On an error, the scratch org DB model should be deleted (or at least flagged and omitted from future API calls).